rasa/server.py

```diff
@@ -0,0 +1,773 @@
+            jwt_data = request.app.auth.extract_payload(request)
+            user = jwt_data.get("user", {})
+
+            username = user.get("username", None)
+            role = user.get("role", None)
+
+            if role == "admin":
+                return True
+            elif role == "user":
+                conversation_id = conversation_id_from_args(args, kwargs)
+                return conversation_id is not None and username == conversation_id
+            else:
+                return False
+
+        @wraps(f)
+        async def decorated(request: Request, *args: Any, **kwargs: Any) -> Any:
+
+            provided = request.args.get("token", None)
+
+            # noinspection PyProtectedMember
+            if token is not None and provided == token:
+                result = f(request, *args, **kwargs)
+                if isawaitable(result):
+                    result = await result
+                return result
+            elif app.config.get("USE_JWT") and request.app.auth.is_authenticated(
+                request
+            ):
+                if sufficient_scope(request, *args, **kwargs):
+                    result = f(request, *args, **kwargs)
+                    if isawaitable(result):
+                        result = await result
+                    return result
+                raise ErrorResponse(
+                    403,
+                    "NotAuthorized",
+                    "User has insufficient permissions.",
+                    help_url=_docs("/server.html#security-considerations"),
+                )
+            elif token is None and app.config.get("USE_JWT") is None:
+                # authentication is disabled
+                result = f(request, *args, **kwargs)
+                if isawaitable(result):
+                    result = await result
+                return result
+            raise ErrorResponse(
+                401,
+                "NotAuthenticated",
+                "User is not authenticated.",
+                help_url=_docs("/server.html#security-considerations"),
+            )
+
+        return decorated
+
+    return decorator
+
+
+def event_verbosity_parameter(request, default_verbosity):
+    event_verbosity_str = request.raw_args.get(
+        "include_events", default_verbosity.name
+    ).upper()
+    try:
+        return EventVerbosity[event_verbosity_str]
+    except KeyError:
+        enum_values = ", ".join([e.name for e in EventVerbosity])
+        raise ErrorResponse(
+            400,
+            "BadRequest",
+            "Invalid parameter value for 'include_events'. "
+            "Should be one of {}".format(enum_values),
+            {"parameter": "include_events", "in": "query"},
+        )
+
+
+def obtain_tracker_store(agent: "Agent", conversation_id: Text):
+    tracker = agent.tracker_store.get_or_create_tracker(conversation_id)
+    if not tracker:
+        raise ErrorResponse(
+            409,
```

Comment:
isn't this rather a 404?

Reply:
I guess you can argue about that. The agent is already loaded, when this request was made. So the error does not come from the request itself, but rather from the time when the agent was loaded. Thus, I would say it is a 409.